app/views/play/cinematic/Loader.js

```diff
@@ -27,45 +26,97 @@ export default class Loader {
    * @returns {Cinematic} The raw JSON object with Cinematic data.
    */
   async loadAssets () {
-    this.data = await get(this.slug)
     this.loadThangTypes(this.data.shots)
+    this.loadPlayerThangType()
+    this.loadBackgroundObjects(this.data.shots)
     await this.load()
     return this.data
   }
 
+  /**
+   * Loads the player thangType from the global `me` object if accessible.
+   * Has a side effect of storing the players thangType by original as a resource.
+   */
+  loadPlayerThangType () {
+    if ((me || {}) && !me.get('heroConfig')) {
+      return
+    }
+    const original = me.get('heroConfig').thangType
+    if (!original) {
+      return
+    }
+
+    this.loadingThangTypes.set(
+      original,
+      (async () => {
```

Comment:
Do we mind seeing this syntax? In this context it could instead  be a `new Promise(() => ... function here ...)`